File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/model/helper/AddFavouritesAsyncTask.java
package com.dbottillo.mtgsearchfree.model.helper;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.dbottillo.mtgsearchfree.model.database.CardsInfoDbHelper;
import com.dbottillo.mtgsearchfree.model.database.FavouritesDataSource;
import com.dbottillo.mtgsearchfree.model.database.MTGCardDataSource;
import com.dbottillo.mtgsearchfree.model.database.MTGDatabaseHelper;
import com.dbottillo.mtgsearchfree.model.MTGCard;

import java.util.ArrayList;
import java.util.List;

public class AddFavouritesAsyncTask extends AsyncTask<String, Void, ArrayList<Object>> {

    private boolean error = false;
    private Context context;

    public AddFavouritesAsyncTask(Context context) {
        this.context = context;
    }

    @Override
    protected ArrayList<Object> doInBackground(String... params) {
        ArrayList<Object> result = new ArrayList<>();

        MTGDatabaseHelper databaseHelper = MTGDatabaseHelper.getInstance(context);
        CardsInfoDbHelper cardsInfoDbHelper = CardsInfoDbHelper.getInstance(context);
        MTGCardDataSource mtgCardDataSource = new MTGCardDataSource(databaseHelper);

        FavouritesDataSource.clear(cardsInfoDbHelper.getWritableDatabase());
        List<MTGCard> cards = mtgCardDataSource.getRandomCard(600);
        for (MTGCard card : cards) {
            FavouritesDataSource.saveFavourites(cardsInfoDbHelper.getWritableDatabase(), card);
        }

        return result;
    }

    @Override
    protected void onPostExecute(ArrayList<Object> result) {
        if (error) {
            Toast.makeText(context, "error", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(context, "finished", Toast.LENGTH_SHORT).show();
        }
    }

}

File: feature_debug/src/main/kotlin/com/dbottillo/mtgsearchfree/debug/AddFavouritesAsyncTask.kt
package com.dbottillo.mtgsearchfree.debug

import android.content.Context
import android.os.AsyncTask
import android.widget.Toast
import com.dbottillo.mtgsearchfree.database.CardDataSource
import com.dbottillo.mtgsearchfree.database.CardsInfoDbHelper
import com.dbottillo.mtgsearchfree.database.FavouritesDataSource
import com.dbottillo.mtgsearchfree.database.MTGCardDataSource
import com.dbottillo.mtgsearchfree.database.MTGDatabaseHelper
import com.google.gson.Gson
import java.lang.ref.WeakReference

/*
    This class is used only on debug to generate random favourites cards
 */
@Suppress("MagicNumber")
internal class AddFavouritesAsyncTask(context: Context) : AsyncTask<String, Void, ArrayList<Any>>() {

    private val context: WeakReference<Context> = WeakReference(context)

    override fun doInBackground(vararg params: String): ArrayList<Any> {
        val result = ArrayList<Any>()

        context.get()?.let {
            val databaseHelper = MTGDatabaseHelper(it)
            val cardsInfoDbHelper = CardsInfoDbHelper(it)
            val cardDataSource = CardDataSource(cardsInfoDbHelper.writableDatabase, Gson())
            val mtgCardDataSource = MTGCardDataSource(databaseHelper.readableDatabase, cardDataSource)

            val favouritesDataSource = FavouritesDataSource(cardsInfoDbHelper.writableDatabase, cardDataSource)
            favouritesDataSource.clear()

            mtgCardDataSource.getRandomCard(60).forEach {
                favouritesDataSource.saveFavourites(it)
            }

            cardsInfoDbHelper.close()
            databaseHelper.close()
            return result
        } ?: return arrayListOf()
    }

    override fun onPostExecute(result: ArrayList<Any>) {
        context.get()?.let {
            Toast.makeText(it, "finished", Toast.LENGTH_SHORT).show()
        }
    }
}
